recipes/physx/4.1.1/conanfile.py

```diff
@@ -0,0 +1,277 @@
+        options_build_type = str(self.options.build_type)
+        if options_build_type == "debug":
+            if settings_build_type != "Debug":
+                raise ConanInvalidConfiguration("Settings build_type=%s is not compatible " \
+                                                "with physx:build_type=%s" % (settings_build_type, options_build_type))
+        elif settings_build_type == "Debug":
+            raise ConanInvalidConfiguration("Settings build_type=%s is not compatible " \
+                                            "with physx:build_type=%s" % (settings_build_type, options_build_type))
+
+        if compiler == "Visual Studio":
+            runtime = str(self.settings.compiler.runtime)
+            if options_build_type == "debug":
+                if runtime != "MDd" and runtime != "MTd":
+                    raise ConanInvalidConfiguration("Visual Studio Compiler runtime MDd or MTd " \
+                                                    "is required when physx:build_type=%s" % options_build_type)
+            elif runtime != "MD" and runtime != "MT":
+                raise ConanInvalidConfiguration("Visual Studio Compiler runtime MD or MT " \
+                                                "is required when physx:build_type=%s" % options_build_type)
+
+    def config_options(self):
+        del self.options.fPIC # fpic is managed by physx build process (in physx/source/compiler/cmake/CMakeLists.txt)
+        if self.settings.os != "Windows":
+            del self.options.enable_float_point_precise_math
+
+    def package_id(self):
+        self.info.settings.build_type = str(self.options.build_type)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        url = self.conan_data["sources"][self.version]["url"]
+        extracted_dir = "PhysX-" + os.path.splitext(os.path.basename(url))[0]
+        try:
+            os.rename(extracted_dir, self._source_subfolder)
+        except:
+            # workaround for permission denied on windows
+            time.sleep(10)
+            os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def _configure_cmake(self):
+        cmake = CMake(self, build_type=str(self.options.build_type))
+
+        if self.settings.os == "Windows":
+            cmake.definitions["TARGET_BUILD_PLATFORM"] = "windows"
+        elif self.settings.os == "Linux":
+            cmake.definitions["TARGET_BUILD_PLATFORM"] = "linux"
+        elif self.settings.os == "Macos":
+            cmake.definitions["TARGET_BUILD_PLATFORM"] = "mac"
+        elif self.settings.os == "Android":
+            cmake.definitions["TARGET_BUILD_PLATFORM"] = "android"
+        elif self.settings.os == "iOS":
+            cmake.definitions["TARGET_BUILD_PLATFORM"] = "ios"
```

Comment:
I would prefer to define a helper method to extract the platform:
```suggestion
def _target_build_platform(self):
        return {
            "Windows" : "windows",
            "Linux" : "linux",
            "Macos" : "mac",
            "Android" : "android",
            "iOS" : "ios"
        }.get(str(self.settings.os))

     def _configure_cmake(self):
         cmake = CMake(self, build_type=str(self.options.build_type))
         cmake.definitions["TARGET_BUILD_PLATFORM"] = self._target_build_platform
```

Reply:
done